From 5628ad8ec199db32f4bdd652dc6956730a67d22c Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Mon, 5 Sep 2016 10:26:51 -0400
Subject: [PATCH] fix installed system boot from encrypt fs failed

While using cryptsetup to encrypt filesystem, it failed to boot
installed system.
-----------------
dracut: luksOpen /dev/sda3 luks-9cf2d2d9-849e-4ecb-a7f8-5918259306d6
Enter passphrase for /dev/sda3: Switched to clocksource tsc

device-mapper: table: 253:0: crypt: Error allocating crypto tfm
device-mapper: ioctl: error adding target to table
device-mapper: reload ioctl on  failed: No such file or directory
Failed to setup dm-crypt key mapping for device /dev/sda3.
Check that kernel supports aes-xts-plain64 cipher (check syslog for more info).
Wrong password
-----------------

The reason is the initramfs (generated by dracut) missed kernel module
xts-aes-aesni.

Config dracut to let initramfs has the kernel driver.

Upstream-Status: Inappropriate [wrlinux specific]

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 pyanaconda/install.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/pyanaconda/install.py b/pyanaconda/install.py
index 1973460..ba3a4f8 100644
--- a/pyanaconda/install.py
+++ b/pyanaconda/install.py
@@ -116,6 +116,8 @@ def doConfiguration(storage, payload, ksdata, instClass):
         ksdata.addons.execute(storage, ksdata, instClass, u)
 
     with progress_report(_("Generating initramfs")):
+        with open(iutil.getSysroot() + "/etc/dracut.conf.d/cryptsetup.conf", "w") as fobj:
+            fobj.write('add_drivers+=" xts-aes-aesni"\n')
         payload.recreateInitrds(force=True)
 
     if willRunRealmd:
-- 
2.8.1

